doc/debugging.rst

```diff
@@ -14,33 +14,88 @@ This allows breakpoints and the execution flow to be viewed and controlled right
 Debugging with ipdb
 ===================
 
-Spyder offers the following debugging features integrated into the native GUI:
+You can have full GUI control over debugger execution from the :guilabel:`Debug` menu, :guilabel:`Debug toolbar` and via configurable keyboard shortcuts, along with the standard ``ipdb`` `console commands`_.
 
-* Multiple means of setting and clearing normal and conditional breakpoints for any line in a file opened in the :doc:`editor`.
+.. _console commands: https://pythonconquerstheuniverse.wordpress.com/2009/09/10/debugging-in-python/
 
-  * By selecting the respective option from the Debug menu.
-  * Through pressing a configurable keyboard shortcut (:kbd:`F12` for normal, or :kbd:`Shift-F12` for conditional breakpoints by default).
-  * By double-clicking to the left of the line number in an open file.
-  * With an ``ipdb.set_trace()`` statement in your code (after ``import`` ing ``pdb``).
-  * Interactively, using the ``b`` command in an ``ipdb`` session.
+GIF: Inset of Spyder's Editor, with a breakpoint set and the condition dialog open (commands from the toolbar)
+
+Spyder's debugger has features like syntax highlighting, completion, and history which work exactly like the IPython console.
+To trigger autocomplete suggestions, press :kbd:`Tab`, and use the up and down arrows to recall previous commands.
+
+GIF: Browsing autocomplete suggestions
+
+You can also take advantage of IPython's `magic functions`_ in debugging mode which will allow you to use ``%matplotlib`` to switch between the ``qt5`` and ``inline`` plotting backends, and ``%timeit`` to check how fast a given snippet of code is.
+
+.. _magic functions: https://ipython.readthedocs.io/en/stable/interactive/magics.html
+
+SCREENSHOT: (timeit)
+
+Additionally the debugger allows highlighting of the current frame (debugging step) in the :doc:`editor`.
+
+SCREENSHOT of highlighting step (close up)
+
+Finally, you can write multiline statements while debugging which will allow you execute code while debugging easily.
+
+GIF of multiline debugging
+
+
+
+===========
+Breakpoints
+===========
+
+Spyder's debugger is integrated with a :guilabel:`Breakpoints` pane, listing the file, line, and condition (if any) of every breakpoint defined (:menuselection:`Debug --> List breakpoints`, or :kbd:`Ctrl-Shift-B` by default).
 
 .. image:: images/debugging/breakpoints-standard.png
    :alt: Spyder's Breakpoints panel, with a number of examples showing file, line number and an optional condition
 
-* A :guilabel:`Breakpoints` pane, listing the file, line, and condition (if any) of every breakpoint defined (:menuselection:`&Debug --> List breakpoints`, or :kbd:`Ctrl-Shift-B` by default).
+There are multiple means of setting and clearing normal and conditional breakpoints for any line in a file opened in the :doc:`editor`.
```

Comment:
```suggestion
There are several different ways to set and clear breakpoints:
```

Greatly simplify my original overly complex, run-on sentence and make it more action- and reader-focused, and remove extraneous material ("any line in the editor")